### DISCUSSION TITLE
Help getting GPIO work for Zephyr and nRF52840

### DISCUSSION THREAD

**Discussion by ujur007** [zephyrproject-rtos/zephyr#58801] [Q&A]
I am struggling on how to use GPIO pins other than what are defined since last two days. At last, I am posting this discussions.
So, I want custom pin GPIO to attach LED or Buzzer. 
What I tried so far is to change the .dts file and add some custom definition like the following.
```
buzzer,user {
        buzzer1: buzzer_1 {
            gpioee = <&gpio1 9 GPIO_ACTIVE_HIGH>;
            label = "Buzzer custom";
        };
    };
```
The in my app code I tried to get the device struct using following code. But I could not get the required device. I tried some different syntax and settings but honestly I do not get a clue how it works.

```
#define BUZZER DT_PATH(buzzer_user)
const struct gpio_dt_spec buzzer = GPIO_DT_SPEC_GET(BUZZER, gpioee);
```

**Comment by vChavezB** [comment]
If you do not know about the device tree macros I recommend that you check this video 

https://www.youtube.com/watch?v=w8GgP3h0M8M

**Comment by ujur007** [comment]
Sorry but this was solved and there was some mistake in my code in the device tree sections. Thanks for posting. you can mark it solved.
